projects/plugins/boost/app/modules/class-module.php

```diff
@@ -56,6 +59,28 @@ final public function is_initialized() {
 		return $this->is_initialized;
 	}
 
+	/**
+	 * Run actions on plugin deactivation.
+	 *
+	 * Override this method in a specific module class to run deactivation
+	 * tasks hooked to jetpack_boost_deactivate action.
+	 *
+	 * @see Module::__construct()
+	 * @see Jetpack_Boost::deactivate()
+	 */
+	public function on_deactivate() {}
+
+	/**
+	 * Run actions on plugin uninstall.
+	 *
+	 * Override this method in a specific module class to run uninstall
+	 * tasks hooked to jetpack_boost_uninstall action.
+	 *
+	 * @see Module::__construct()
+	 * @see Jetpack_Boost::uninstall()
+	 */
+	public function on_uninstall() {}
```

Comment:
I am a bit curious, if we could make these abstract or is there a better way to provide those functions without defining empty function.

Reply:
If we make these methods as abstracts, then we are forcing the extending modules to define these functions. I do not see a necessity for that since an extending module may not want to run any task at the time of deactivation or uninstall. `method_exists` can be another way. I think it will break the code consistency though. What do you think?

Reply:
Yeah, I think empty default implementations here is the way to go.